### PR TITLE
[ios] Refactor formatters

### PR DESCRIPTION
**Pull request by kirylkaveryn**
This PR refactors the formatters in iOS:
1. Creates 3 formatters: for distances, altitudes and date/times that call the core formatters or use the shared static instance to format the values because the formatter initialization is expensive [resource-intensive](https://sarunw.com/posts/how-expensive-is-dateformatter/).
2. Removes unnecessary instantiations of formatters (there were a lot)

### Screenshots:
Before/After km

image image

Before/After miles

image image

### CODE REVIEW

File: iphone/CoreApi/CoreApi/Formatting/DistanceFormatter.swift [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import Foundation
+
+// TODO: Replace this class with a core's implementation
```
Comment (vng) [inline iphone/CoreApi/CoreApi/Formatting/DistanceFormatter.swift:3]: Did you check platform::Distance ?

File: iphone/CoreApi/CoreApi/Formatting/AltitudeFormatter.swift [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import Foundation
+
+// TODO: Replace this class with a core's implementation
```
Comment (vng) [inline iphone/CoreApi/CoreApi/Formatting/AltitudeFormatter.swift:3]: Did you check platform::Distance::FormatAltitude ?

File: iphone/CoreApi/CoreApi/Formatting/DistanceFormatter.swift (hunk 1 above)
Comment (kirylkaveryn) [reply iphone/CoreApi/CoreApi/Formatting/DistanceFormatter.swift]: Thanks! Replaced with the core implementation.

File: iphone/CoreApi/CoreApi/Formatting/AltitudeFormatter.swift (hunk 2 above)
Comment (kirylkaveryn) [reply iphone/CoreApi/CoreApi/Formatting/AltitudeFormatter.swift]: fixed

**Comment by kirylkaveryn** [comment]
@vng I update the Distance and Altitude formatetrs to call the platform's impl.

**Review by vng — APPROVED** [review]

**Pull request merged**